oauth2_lib/fastapi.py

```diff
@@ -168,11 +175,7 @@ def __init__(
         resource_server_id: str,
         resource_server_secret: str,
         oidc_user_model_cls: type[OIDCUserModel],
-        id_token_extractor: IdTokenExtractor | None = None,
     ):
-        if not id_token_extractor:
-            self.id_token_extractor = HttpBearerExtractor()
```

Comment:
Maybe keep the kwarg in `__init__` and when a value is passed log a (deprecation) warning instead?

Reply:
Great spot by you (@Igoranze): if a custom `id_token_extractor` is passed then it's not actually assigned to `self.id_token_extractor` so `authenticate()` raises an `AttributeError` when trying to call it.

The only way this can have worked for others is if they override `authenticate()` as well. So, we don't need to worry about backwards compatibility